python/nmodl/ode.py

```diff
@@ -643,15 +643,27 @@ def differentiate2c(expression, dependent_var, vars, prev_expressions=None):
     # differentiate w.r.t. x
     diff = expr.diff(x).simplify()
 
+    # could be something generic like f'(x), in which case we use finite differences
+    if needs_finite_differences(diff):
+        diff = (
+            transform_expression(diff, discretize_derivative)
+            .subs({finite_difference_step_variable(x): 1e-3})
```

Comment:
That looks very specific to CVODE, maybe one can delay picking the step size. Okay, if it's a one-sided difference double check that it's not because of side-effects. I've seen the pattern:
```
i1 = f(v + dv);
i0 = f(v);
didv = (i1 - i0)/dv;
```
This is done to get `f(v)` after `f(v + dv)` because `f(v)` sets state (like `dX` variables) and it looked like that state should be for `v` not `v + dv`.

Reply:
Added `stepsize` param (with a default of `1e-3`) to keep it general.